q2_diversity/_alpha/_method.py

```diff
@@ -36,12 +36,15 @@ def alpha_phylogenetic(table: biom.Table, phylogeny: skbio.TreeNode,
     sample_ids = table.ids(axis='sample')
     feature_ids = table.ids(axis='observation')
 
-    result = skbio.diversity.alpha_diversity(
-                metric=metric,
-                counts=counts,
-                ids=sample_ids,
-                otu_ids=feature_ids,
-                tree=phylogeny)
+    try:
+        result = skbio.diversity.alpha_diversity(metric=metric,
+                                                 counts=counts,
+                                                 ids=sample_ids,
+                                                 otu_ids=feature_ids,
+                                                 tree=phylogeny)
+    except skbio.tree.MissingNodeError:
+        raise skbio.tree.MissingNodeError("All ``feature_ids`` must be "
```

Comment:
From what I can tell we can possibly have a few variants of the `MissingNodeError` raised here (@jairideout correct me if I am mistaken). If this is the case, we probably want to check the error message itself and make sure we are replacing the right one.